src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,291 @@
+
+            const token = getTokenAtPosition(sourceFile, context.span.start);
+            const name = token.getText();
+            const allActions: CodeAction[] = [];
+
+            // Get existing ImportDeclarations from the source file
+            const imports: ImportDeclaration[] = [];
+            if (sourceFile.statements) {
+                for (const statement of sourceFile.statements) {
+                    if (statement.kind === SyntaxKind.ImportDeclaration) {
+                        imports.push(<ImportDeclaration>statement);
+                    }
+                }
+            }
+
+            // Check if a matching symbol is exported by any ambient modules that has been declared
+            const ambientModules = checker.getAmbientModules();
+            for (const moduleSymbol of ambientModules) {
+                const exports = checker.getExportsOfModule(moduleSymbol) || [];
+                for (const exported of exports) {
+                    if (exported.name === name) {
+                        allActions.push(getCodeActionForImport(removeQuotes(moduleSymbol.getName())));
+                    }
+                }
+            }
+
+            // Check if a matching symbol is exported by any files known to the compiler
+            for (const file of allFiles) {
+                const exports = file.symbol && file.symbol.exports;
+                if (exports) {
+                    for (const exported in exports) {
+                        if (exported === name) {
+                            let moduleSpecifier: string;
+                            const sourceDir = getDirectoryPath(sourceFile.fileName);
+                            if (file.fileName.indexOf(nodeModulesDir) !== -1) {
+                                moduleSpecifier = convertPathToModuleSpecifier(file.fileName);
+                            }
+                            else {
+                                // Try and convert the file path into one relative to the source file
+
+                                const pathName = getRelativePathToDirectoryOrUrl(
+                                    sourceDir,
+                                    file.fileName,
+                                    sourceDir,
+                                    createGetCanonicalFileName(useCaseSensitiveFileNames),
+                                    /* isAbsolutePathAnUrl */ false
+                                );
+
+                                // Make sure we got back a path that can be a valid module specifier
+                                const isRootedOrRelative = isExternalModuleNameRelative(pathName) || isRootedDiskPath(pathName);
+                                moduleSpecifier = removeFileExtension(isRootedOrRelative ? pathName : combinePaths(".", pathName));
+                            }
+
+                            allActions.push(getCodeActionForImport(moduleSpecifier, (a, b) =>

```

Comment:
I think we should split this into phases, first phase is to find all [exported symbol, module] that match the name we are looking for, then the next phase is to generate the action for each symbol.